src/main/java/org/peergos/blockstore/SQLiteBlockstore.java

```diff
@@ -0,0 +1,104 @@
+package org.peergos.blockstore;
+
+import io.ipfs.cid.Cid;
+import io.ipfs.multihash.Multihash;
+import org.peergos.Hash;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.sql.*;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.logging.Logger;
+
+public class SQLiteBlockstore implements Blockstore {
+    private static final Logger LOG = Logger.getLogger(FileBlockstore.class.getName());
+
+    final String pathToSQLiteFile;
+    final String BLOCK_STORE_TABLE = "blocks";
+
+    public SQLiteBlockstore(String pathToSQLiteFile) {
+        if (pathToSQLiteFile == null) {
+            throw new IllegalArgumentException("filePath cannot be NULL");
+        }
+        this.pathToSQLiteFile = pathToSQLiteFile;
+        createTable();
+        LOG.info("Using SQLiteBlockstore at location: " + pathToSQLiteFile);
+    }
+
+    private Connection getConnection() throws SQLException {
+        return DriverManager.getConnection("jdbc:sqlite:" + pathToSQLiteFile);
+    }
+
+    private void createTable() {
+        String createSQL = "create table if not exists " + BLOCK_STORE_TABLE + " (hash text primary key not null, data blob not null);";
+        try (Connection connection = getConnection();
+             PreparedStatement select = connection.prepareStatement(createSQL)) {
+            select.execute();
+        } catch (SQLException ex) {
+            throw new IllegalStateException(ex);
+        }
+    }
+
+    @Override
+    public CompletableFuture<Boolean> has(Cid c) {
+        String selectSQL = "SELECT 1 FROM " + BLOCK_STORE_TABLE + " WHERE hash=?";
+        try (Connection connection = getConnection();
+             PreparedStatement pstmt = connection.prepareStatement(selectSQL)) {
+            pstmt.setString(1, c.toString());
```

Comment:
We should enforce the encoding we use for cid here, or use the byte encoding

Reply:
SQLite block store removed for now due to size of sqlite jar file